src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -1791,6 +1791,13 @@ impl Display for Decimal {
     }
 }
 
+impl crate::coprocessor::codec::data_type::AsMySQLBool for Decimal {
+    #[inline]
+    fn as_mysql_bool(&self) -> bool {
+        self.as_f64().unwrap_or(0.0).round() != 0f64
```

Comment:
We should return an error if it happens.

Reply:
I have fixed it in upstream large PR. There is a lot of diff, shall we merge this PR first (since the interface is not used at all, the incorrectness doesn't matter), then before it is really used I will send another PR to fix it?

Reply:
OK